libcontainer/specconv/spec_linux.go

```diff
@@ -467,6 +516,28 @@ func CreateLibcontainerConfig(opts *CreateOpts) (*configs.Config, error) {
 				MemBwSchema:   spec.Linux.IntelRdt.MemBwSchema,
 			}
 		}
+		if spec.Linux.MemoryPolicy != nil &&
+			(spec.Linux.MemoryPolicy.Mode != "" ||
+				spec.Linux.MemoryPolicy.Nodes != "" ||
+				len(spec.Linux.MemoryPolicy.Flags) > 0) {
```

Comment:
Can this be written in a more straightforward way?

```go
if spec.Linux.MemoryPolicy != nil {
	if specMp.Mode != "" {
		// set confMp.Mode
	}
	if specMp.Nodes != "" {
		// set confMp.Nodes
	}
	if len(specMp.Flags) > 0 {
		// set confMp.Mode flags
	}
}
```

Reply:
To improve readability, I rewrote this and added a helper function for emptiness check of spec.Linux.MemoryPolicy.

I want to handle completely empty MemoryPolicy equivalent of nil, but if a MemoryPolicy includes nodes and/or flags but empty Mode, I want to error out with "invalid mode".